src/node/mod.rs

```diff
@@ -727,15 +727,18 @@ impl Node {
                         *msg.dst_key(),
                         *message.clone(),
                     ),
-                Variant::BouncedUnknownMessage {
-                    message,
-                    parsec_version,
-                } => stage.handle_bounced_unknown_message(
-                    &mut self.core,
-                    msg.src().to_sender_node(sender)?,
-                    message.clone(),
-                    *parsec_version,
-                ),
+                Variant::BouncedUnknownMessage(message) => {
+                    if let Some(sender_last_key) = msg.dst_key() {
+                        stage.handle_bounced_unknown_message(
+                            &mut self.core,
+                            msg.src().to_sender_node(sender)?,
+                            message.clone(),
+                            sender_last_key,
+                        )
+                    } else {
+                        unreachable!()
```

Comment:
This is not really unreachable. A malicious/faulty peer can send us a message with missing `dst_key` and we would hit this branch and crash. Just log something and return `Err(InvalidMessage)` or something.